final_task/rss_reader/articles_handler.py

```diff
@@ -0,0 +1,226 @@
+
+
+def parse_rss(rss_url: str) -> feedparser.FeedParserDict:
+    """Parse the rss"""
+    return feedparser.parse(rss_url)
+
+
+def create_rss_json(data: List[single_article.SingleArticle]):
+    """Prints result as JSON in stdout"""
+    return json.dumps(data, cls=single_article.EnhancedJSONEncoder, ensure_ascii=False)
+
+
+def print_article(data: single_article.SingleArticle) -> None:
+    """Prints a single article in stdout"""
+    print("-" * 120)
+    print(data)
+    print("-" * 120)
+
+
+def print_rss_articles(rss_articles: List[single_article.SingleArticle]) -> None:
+    """Prints source feed and articles"""
+    for article in rss_articles:
+        print_article(article)
+
+
+def find_links_in_article(article: feedparser.FeedParserDict) -> Tuple[List, str]:
+    """Finds images and links in the article"""
+    all_links = []
+
+    soup = BeautifulSoup(article['summary'], features="html.parser")
+    for index, img in enumerate(soup.find_all('img'), 1):
+        if (str(img['src']) == '') and (str(img['alt']) == ''):
+            continue
+        if str(img['alt']) == '':
+            all_links.append([str(img['src']), 'image'])
+            img.replaceWith(f"[image {index}: no description][{index}] ")
+        elif str(img['src']) == '':
+            img.replaceWith(f"[image(no link): {img['alt']}] ")
+        else:
+            all_links.append([str(img['src']), 'image'])
+            img.replaceWith(f"[image {index}: {img['alt']}][{index}] ")
+
+    for link in article['links']:
+        if link['href'] not in all_links:
+            all_links.append([link['href'], 'other'])
+            """if link['href'] == article['link']:
+                all_links.append([link['href'], 'text'])
+            else:
+                head = requests.head(link['href'])
+                header = head.headers
+                content_type = header.get('content-type')
+
+                if 'image' in content_type:
+                    all_links.append([link['href'], 'image'])
+                else:
+                    all_links.append([link['href'], 'text'])"""
+
+    return all_links, str(soup.text)
+
+
+def unescape(text_to_unescape: str) -> str:
+    """Unescape text"""
+    return html.unescape(text_to_unescape)
+
+
+def is_connected():
+    hostname = 'https://www.google.com/'
+    try:
+        urllib.request.urlopen(hostname)
+        return True
+    except urllib.request.HTTPError:
+        return False
+    except urllib.request.URLError:
+        return False
+
+
+def get_articles(parsed: feedparser.FeedParserDict, limit: int) -> List[single_article.SingleArticle]:
+    """Returns list with articles."""
+    articles = []
+    feed = parsed['feed']
+    entries = parsed['entries']
+    logging.info(f'There is {len(entries)} entries')
+
+    if limit is None:
+        limit = len(entries)
+
+    try:
+        for index, entry in enumerate(entries):
+            if index == limit:
```

Comment:
Consider using `slice` instead of comparing limit with index value on every iteration